src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -451,15 +451,28 @@ async def resolve_hostnames(self, hostnames):
         hostnames = list(set(hostnames))
 
         async def resolve(hostname):
-            if domain := get_wildcard_domain(hostname):
-                hostname = domain
-
             try:
-                return (
-                    await asyncio.wait_for(self.middleware.run_in_thread(socket.getaddrinfo, hostname, None), 5)
-                )[0][4][0]
+                try:
+                    # If this is an IP address, just return it
+                    ip = IpAddress()
+                    ip(hostname)
+                    return hostname
+                except ValueError:
+                    # Not an IP address, should be a name
+                    if domain := get_wildcard_domain(hostname):
+                        hostname = domain
+
+                    if leftmost_has_wildcards(hostname):
+                        # We know this will not resolve
+                        return None
+                    else:
+                        try:
+                            return (await self.middleware.call('dnsclient.forward_lookup', {"names": [hostname]}))[0]['address']
```

Comment:
This API only allows searching one record type per query (defaulting with A), do we ned to be concerned about AAAA?